src/converters/light.ts

```diff
@@ -0,0 +1,348 @@
+  private cached_saturation = 0.0;
+  private received_saturation = false;
+
+  constructor(expose: ExposesEntryWithFeatures, private readonly accessory: AbstractAccessory) {
+    const endpoint = expose.endpoint;
+    this.identifier = LightHandler.generateIdentifier(endpoint);
+
+    const features = expose.features.filter(e => exposesHasProperty(e) && !accessory.isPropertyExcluded(e.property))
+      .map(e => e as ExposesEntryWithProperty);
+
+    // On/off characteristic (required by HomeKit)
+    const potentialStateExpose = features.find(e => e.name === 'state' && exposesIsPublished(e) && exposesCanBeSet(e));
+    if (potentialStateExpose === undefined || !exposesHasBinaryProperty(potentialStateExpose)) {
+      throw new Error('Required "state" property not found for Light.');
+    }
+    this.stateExpose = potentialStateExpose;
+
+    let serviceName = accessory.displayName;
+    if (endpoint !== undefined) {
+      serviceName += ' ' + endpoint;
+    }
+
+    accessory.log.debug(`Configuring Light for ${serviceName}`);
+    const service = accessory.getOrAddService(new hap.Service.Lightbulb(serviceName, endpoint));
+
+    getOrAddCharacteristic(service, hap.Characteristic.On).on('set', this.handleSetOn.bind(this));
+    const onOffValues = new Map<CharacteristicValue, CharacteristicValue>();
+    onOffValues.set(this.stateExpose.value_on, true);
+    onOffValues.set(this.stateExpose.value_off, false);
+    this.monitors.push(new MappingCharacteristicMonitor(this.stateExpose.property, service, hap.Characteristic.On,
+      onOffValues));
+
+    // Brightness characteristic
+    this.tryCreateBrightness(features, service);
+
+    // Color temperature
+    this.tryCreateColorTemperature(features, service);
+
+    // Color: Hue/Saturation or X/Y
+    this.tryCreateColor(expose, service, accessory);
+  }
+
+  identifier: string;
+  get getableKeys(): string[] {
+    const keys: string[] = [];
+    if (exposesCanBeGet(this.stateExpose)) {
+      keys.push(this.stateExpose.property);
+    }
+    if (this.brightnessExpose !== undefined && exposesCanBeGet(this.brightnessExpose)) {
+      keys.push(this.brightnessExpose.property);
+    }
+    if (this.colorTempExpose !== undefined && exposesCanBeGet(this.colorTempExpose)) {
+      keys.push(this.colorTempExpose.property);
+    }
+    if (this.colorExpose !== undefined && this.colorExpose.property !== undefined) {
+      if ((this.colorComponentAExpose !== undefined && exposesCanBeGet(this.colorComponentAExpose))
+        || (this.colorComponentBExpose !== undefined && exposesCanBeGet(this.colorComponentBExpose))
+      ) {
+        keys.push(this.colorExpose.property);
+      }
+    }
+    return keys;
+  }
+
+  updateState(state: Record<string, unknown>): void {
+    this.monitors.forEach(m => m.callback(state));
+  }
+
+  private tryCreateColor(expose: ExposesEntryWithFeatures, service: Service, accessory: AbstractAccessory) {
+    // First see if color_hs is present
```

Comment:
Not sure if `color_hs` should be preferred over `color_xy` or the other way around.
Given that (as far as I can tell) most colored lights at least support `color_xy`, it might make more sense to prefer that one instead.